### PR TITLE
Check the variable before iteration

### PR DESCRIPTION
**Pull request by 8pattern**
## Description

The assertion only works in debug mode. So if the *mappedKeys* is null, it will raise an error in release mode.

## Related Issues

None

## Tests

I added the following tests:

None

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/services/raw_keyboard.dart [hunk 1]
```diff
@@ -607,8 +607,10 @@ class RawKeyboard {
       assert(mappedKeys != null,
         'Platform key support for ${Platform.operatingSystem} is '
         'producing unsupported modifier combinations.');
-      for (final PhysicalKeyboardKey physicalModifier in mappedKeys) {
-        modifierKeys[physicalModifier] = _allModifiers[physicalModifier];
+      if (mappedKeys != null) {
```
Comment (goderbauer) [inline packages/flutter/lib/src/services/raw_keyboard.dart:610]: There is an assert right above this line saying that mappedKeys should never be null. So, this `if` doesn't appear to be necessary?

/cc @gspencergoog

**Comment by goderbauer** [comment]
Thanks for the contribution. We require that all PRs have tests, can you please add one that shows that this if is actually necessary (see my other comment). Also, you'll have to sign the CLA. Thanks.

File: packages/flutter/lib/src/services/raw_keyboard.dart (hunk 1 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/services/raw_keyboard.dart:610]: If that assert is firing (or mappedKeys is `null` in release mode), then the `_modifierKeyMap` is probably missing a modifier key.

What modifier key is triggering the assert?

**Comment by goderbauer** [comment]
I am going to close this one since the additional `if` check shouldn't be required given the assert just before it. If this causes any problems, please file an issue with the additional information requested by greg in the comment above. Thanks.

**Pull request closed without merging**